### ISSUE TITLE
OSX support

### ISSUE THREAD

**Issue by maximzubkov** [giganticode/codeprep#11]
Hello!

Thank you for your work it is amazing!
My name is Maksim Zubkov, I am an inter at JetBrains Research. I tried to use your tool for BPE tokenization of C++ code and got the following error:
```
OSError: Calculation of vocabulary is not supported on OSX.
```
Could you please explain why it is not currently supported, and what can I do to use your lib on MacOS?

Thank you!

**Comment by hlibbabii** [comment]
Hi, @maximzubkov! Thanks for your kind words and for reporting the issue! The vocabulary is calculated in multiple processes by default (was essential since we used to calculate vocabularies of huge corpora), and the synchronization primitives used were not supported on OSx. I made vocabulary calculation on OSx run in a single process and released version 1.0.4 on PyPI with the fix.

P.S. When writing this tool, the main goal was supporting our research, and I am glad if others can also find it useful (even though the use-cases beyond ours have been tested way less).

**Comment by maximzubkov** [comment]
Greate, thank you!

I installed the 1.0.4 version, and got another error:
```
name 'NonAtomicCounter' is not defined
```
Seems like you have forgotten to import `NonAtomicCounter` for `util`. 
I created pull request with this minor fix, seems like it works on my MacOs https://github.com/giganticode/codeprep/pull/16

**Comment by hlibbabii** [comment]
> Seems like you have forgotten to import `NonAtomicCounter` for `util`.
> I created a pull request with this minor fix, seems like it works on my MacOs #16

Oops, good catch, thank you!
